staging/src/k8s.io/api/networking/v1/types.go

```diff
@@ -52,31 +52,38 @@ const (
 
 // NetworkPolicySpec provides the specification of a NetworkPolicy
 type NetworkPolicySpec struct {
-	// Selects the pods to which this NetworkPolicy object applies. The array of
-	// ingress rules is applied to any pods selected by this field. Multiple network
-	// policies can select the same set of pods. In this case, the ingress rules for
-	// each are combined additively. This field is NOT optional and follows standard
+	// Selects the pods to which this NetworkPolicy object applies for ingress and/or egress,
+	// as determined by `policyTypes`.  The `ingress` and `egress` fields characterize
+	// connections allowed by this policy.
+	// Multiple NetworkPolicy objects can select a given pod, and the connections they
+	// allow are combined additively.
+	// If any NetworkPolicy applies to a given pod for ingress then the allowed incoming
+	// connections are those from the pod's local node and those characterized by an
+	// ingress rule of an applicable policy.
+	// If no NetworkPolicy applies to a given pod for ingress
+	// then all ingress to that pod is allowed.
+	// If any NetworkPolicy applies to a given pod for egress then the allowed outgoing
+	// connections are those characterized by an egress rule of an applicable policy.
+	// If no NetworkPolicy applies to a given pod for egress
+	// then all egress from that pod is allowed.
+	// A connection from some source pod to some destination pod is allowed if egress
+	// from the source is allowed and ingress to the destination is allowed.
+	// This field is NOT optional and follows standard
```

Comment:
the `NOT` optional is a bit wierd.  I would say we usually just say:

> This field is required and follows standard label selector semantics.

"is required" is more correct than "not optional" and easier for a user to understand.

Reply:
done